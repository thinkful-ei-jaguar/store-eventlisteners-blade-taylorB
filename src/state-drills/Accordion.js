import React from 'react';
import App from '../App'

class Accordion extends React.Component {
  
  static defaultProps = {
    sections: []
  }

  state = {
    currentButton: null,
    currentIndex: null
  }
  
  
  handleClick (content, index) {
    this.setState({
      currentButton: content,
      currentIndex: index
    })
  }

  renderContent(index) {
    const currentButton = this.state.currentButton;
    
    if (index === this.state.currentIndex) {
      return (
        <p>{this.state.currentButton}</p>
      )
    } else {
      return;
    }
  }

  renderButtons() {
    return this.props.section.map((item, index) => (
      (<li key={index}>
        <button onClick={() => this.handleClick(item.content, index)}>{item.title}</button>
        {this.renderContent(index)}
      </li>)
    ))
  }

  render () {
    return (
      <ul>
        {this.renderButtons()} 
      </ul>
    )
  }
}

export default Accordion;