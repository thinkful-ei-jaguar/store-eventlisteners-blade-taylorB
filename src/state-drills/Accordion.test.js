import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('Accordion component', () => {


  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<Accordion section={sections} />, div);
    ReactDOM.unmountComponentAtNode(div);

  });

  it('Renders an empty li when sections is not supplied', () => {
    const wrapper = shallow(<Accordion section={[]} />)
    expect(toJson(wrapper)).toMatchSnapshot()

  })

  it('Component renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion section={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  } )

  it('The component opens a clicked section', () => {
    const wrapper = shallow(<Accordion section={sections} />)
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  } )

  it('The component only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion section={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  } )
})