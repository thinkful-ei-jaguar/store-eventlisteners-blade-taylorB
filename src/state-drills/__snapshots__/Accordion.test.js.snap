// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component Renders an empty li when sections is not supplied 1`] = `<ul />`;

exports[`Accordion component The component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion component The component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;
